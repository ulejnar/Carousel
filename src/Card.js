import React from "react";
import "./Card.css";
import image1 from "./image1.jpg";

function Card(props) {
  return (
    <div className="Card">
      <h4 className="Card-title">{props.caption}</h4>
      <img className="Card-image" src={props.src} alt={props.caption} />
      <small className="Card-small">
        Image {props.currNum} of {props.totalNum}.
      </small>
    </div>
  );
}
Card.defaultProps = {
  src: image1,
  caption: "Photo by Richard Pasquarella on Unsplash",
  currNum: 10,
  totalNum: 12
}

export default Card;
