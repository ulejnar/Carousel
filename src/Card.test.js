import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card.js";

it("renders without crashing", function(){
  render (<Card/>)
})

// snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<Card/>);
  expect(asFragment()).toMatchSnapshot();
});