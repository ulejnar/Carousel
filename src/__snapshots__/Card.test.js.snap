// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Photo by Richard Pasquarella on Unsplash
    </h4>
    <img
      alt="Photo by Richard Pasquarella on Unsplash"
      class="Card-image"
      src="image1.jpg"
    />
    <small
      class="Card-small"
    >
      Image 10 of 12.
    </small>
  </div>
</DocumentFragment>
`;
